### PR TITLE
different approach to generating schema

### PR DESCRIPTION
**Pull request by mixmix**
This approach has advantages:
- the `index.js` isn't performing a surprise `fs.writeFile`
- the schema are generated with only the definitions they need
- there is a simple script for updating the `schema.json`
- the `schema.js` is no longer a fragment that could accidentally be used by itself, and being JS it supports comments
- what we export in JS is EXACTLY the same as what other languages would import from `schema.json`

### CODE REVIEW

File: group/add-member/schema.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    },
+    tangles: {
+      type: 'object',
+      required: [
+        'group',
+        'members'
+      ],
+      properties: {
+        group: { $ref: '#/definitions/tangle/update' },
+        members: { $ref: '#/definitions/tangle/update' }
```
Comment (mixmix) [inline group/add-member/schema.js:46]: this used to say `tangle/any` for members... I think this was incorrect

File: group/definitions.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+const groupId = {
+  type: 'string',
+  pattern: '^ssb:identity/group/[a-zA-Z0-9_\\-]{42}[AEIMQUYcgkosw048]=$'
+}
+
+const feedId = {
+  type: 'string',
+  pattern: '^ssb:feed/classic/[a-zA-Z0-9_\\-]{42}[AEIMQUYcgkosw048]=$'
```
Comment (mixmix) [inline group/definitions.js:14]: :fire: this is gonna bite us in the ass when someone wants to make a buttwooo based feed

```suggestion
  pattern: `^ssb:feed/(${validFeeds.join('|')})/[a-zA-Z0-9_\\-]{42}[AEIMQUYcgkosw048]=$`
```

we could do something like this... but maybe it's another PR later

File: group/definitions.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
+const messageId = {
+  type: 'string',
+  $comment: '42 chars + 1 char from a limited subset (last bits are 0) + 1 =',
+  pattern: '^ssb:message/classic/[a-zA-Z0-9_\\-]{42}[AEIMQUYcgkosw048]=$'
+}
+
+const groupId = {
+  type: 'string',
+  pattern: '^ssb:identity/group/[a-zA-Z0-9_\\-]{42}[AEIMQUYcgkosw048]=$'
```
Comment (mixmix) [inline group/definitions.js:9]: :hot_pepper:  this is gonna need to be more flexible for Ahau.... unless this is only needed for publishing new messages

File: group/definitions.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      }
+    ]
+  }
+}
+
+// USAGE:
+// definitions: {
+//   ...definitions.groupId,
+//   ...definitions.tangle.update   // not this pull in everything needed for this defn
+// }
```
Comment (mixmix) [inline group/definitions.js:68]: take advantage of how the spread operator will allow us to de-dup!

File: group/definitions.js (hunk 2 above)
Comment (Powersource) [reply group/definitions.js:14]: yeah i think that's for later

File: group/add-member/schema.js (hunk 1 above)
Comment (Powersource) [reply group/add-member/schema.js:50]: Separate issue. And we might want `any` when doing this issue https://github.com/ssbc/ssb-tribes2/issues/33

Comment (Powersource) [reply group/add-member/schema.js]: Switched it back

**Review by Powersource — APPROVED** [review]
Nice

**Pull request merged**